src/components/CustomGeometry.tsx

```diff
@@ -7,8 +7,12 @@ import { shallowEqual } from 'react-redux'
 import * as THREE from 'three'
 
 const CustomGeometry: FC = () => {
-  const { coorinatesMatrix, indicesMatrix, colors, displacement, displayNodeIndices, display, displacementScale } =
-    useAppSelector((store) => store.model, shallowEqual)
+  const { coorinatesMatrix, indicesMatrix, colors, displacement, display, displacementScale } = useAppSelector(
+    (store) => store.model,
+    shallowEqual
+  )
+
+  const { displayNodeIndices } = useAppSelector((store) => store.modelViewSetting, shallowEqual)
```

Comment:
instead of destructuring the entire `modelViewSetting` object, it's better to directly extract the required `displayNodeIndices` value
we use `shallowEqual` when we're destructuring multiple properties, and we want to optimize comparisons
```suggestion
  const displayNodeIndices  = useAppSelector((store) => store.modelViewSetting.displayNodeIndices)
```